app/Http/Controllers/NotesController.php

```diff
@@ -3,18 +3,125 @@
 namespace OGame\Http\Controllers;
 
 use Illuminate\Http\Request;
+use Illuminate\Http\JsonResponse;
 use Illuminate\View\View;
+use OGame\Services\NoteService;
 
 class NotesController extends OGameController
 {
     /**
      * Shows the notes popup page
      *
      * @param Request $request
+     * @param NoteService $noteService
      * @return View
      */
-    public function overlay(Request $request): View
+    public function overlay(Request $request, NoteService $noteService): View
     {
-        return view('ingame.notes.overlay');
+        $data = [];
+
+        $notesDeleted = false;
+
+        if ($request->isMethod('post')) {
+            $deleteMethod = $request->input('noticeDeleteMethode');
+
+            if ($deleteMethod == 1) {
+                $noteIds = $request->input('delIds', []);
+                if (!empty($noteIds)) {
+                    $noteService->deleteMarkedNotes($noteIds);
+                    $notesDeleted = true;
+                }
+            } elseif ($deleteMethod == 2) {
+                $noteService->deleteAllNotesForUser();
+                $notesDeleted = true;
+            }
+
+            if ($notesDeleted) {
+                $data['success'] = __('Note(s) has(ve) been deleted');
+            }
+        }
+
+        $data['notes'] = $noteService->getAllNotesForUser();
+        return view('ingame.notes.overlay', $data);
+    }
+
+    /**
+     * Shows the notes view popup page
+     *
+     * @param Request $request
+     * @param NoteService $noteService
+     * @return View
+     */
+    public function view(Request $request, NoteService $noteService): View
+    {
+        $note = null;
+
+        $noteId = $request->input('id');
+        if ($noteId) {
+            $note = $noteService->getNoteById($noteId);
+        }
+
+        return view('ingame.notes.create')->with([
+            'noteId' => $note ? $note->id : 0,
+            'priority' => $note ? $note->priority : 2,
+            'subject' => $note ? $note->subject : '',
+            'content' => $note ? $note->content : '',
+        ]);
+    }
+
+    /**
+     * Create a new note
+     *
+     * @param Request $request
+     * @param NoteService $noteService
+     * @return JsonResponse
+     */
+    public function ajaxCreate(Request $request, NoteService $noteService): JsonResponse
+    {
+        $validated = $request->validate([
+            'id' => [
+                'nullable',
+                'integer',
+                function ($attribute, $value, $fail) use ($noteService) {
+                    if ($value !== '0' && $value !== null && !$noteService->noteExistsAndBelongsToUser($value)) {
```

Comment:
Instead of manually checking on `$value !== '0' && $value !== null` you could use the PHP `empty()` check here which does both.